rbd/migration_test.go

```diff
@@ -0,0 +1,174 @@
+//go:build !(octopus || nautilus) && ceph_preview
+// +build !octopus,!nautilus,ceph_preview
+
+package rbd
+
+import (
+	"testing"
+
+	"github.com/ceph/go-ceph/rados"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func createAndWriteDataToImage(t *testing.T, ioctx *rados.IOContext) string {
+	opts := NewRbdImageOptions()
+	defer opts.Destroy()
+
+	assert.NoError(t,
+		opts.SetUint64(ImageOptionOrder, uint64(testImageOrder)))
+
+	name := GetUUID()
+	err := CreateImage(ioctx, name, 1<<25, opts)
+	require.NoError(t, err)
+
+	img, err := OpenImage(ioctx, name, NoSnapshot)
+	require.NoError(t, err)
+
+	img.CreateSnapshot("snap1")
+
+	// write some data
+	_, err = img.WriteAt([]byte("sometimes you feel like a nut"), 0)
+	assert.NoError(t, err)
+
+	err = img.Close()
+	assert.NoError(t, err)
+
+	return name
+}
+
+func TestMigration(t *testing.T) {
+	conn := radosConnect(t)
+
+	pool := GetUUID()
+
+	err := conn.MakePool(pool)
+	assert.NoError(t, err)
+
+	ioctx, err := conn.OpenIOContext(pool)
+	require.NoError(t, err)
+
+	// setup image
+	name := createAndWriteDataToImage(t, ioctx)
+
+	// call MigrationPrepare
```

Comment:
These sorts of comments can be removed. They're great while you're still writing your code as a placeholder but they add little to no value to a reader in the future.

Reply:
Done